### PR TITLE
Register Tensor ops to lazy tensor @open sesame 06/11 17:01

### PR DESCRIPTION
**Pull request by zhoonit**
**Changes proposed in this PR:**
- Register tensor ops to `LazyTensor`
- Add test fixture to `LazyTensor`
- Add tests to `LazyTensor`
- Change `Tensor::apply` signature to permit closures

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: test/unittest/unittest_nntrainer_lazy_tensor.cpp [hunk 1]
```diff
@@ -12,12 +12,70 @@
...
+  nntrainer_LazyTensorOpsTest() {}
+
+  virtual void SetUp() {
+    target = nntrainer::Tensor(batch, channel, height, width);
+    GEN_TEST_INPUT(target, i * (batch * height) + j * (width) + k + 1 + l);
+
+    original.copy(target);
+  }
+
+  // virtual void TearDown()
```
Comment (dongju-chae) [inline test/unittest/unittest_nntrainer_lazy_tensor.cpp:32]: Unnecessary?

File: nntrainer/include/lazy_tensor.h [hunk 2]
```diff
@@ -33,10 +33,97 @@ class LazyTensor {
...
+  /**
+   * @brief     Wrapper method of transpose. see tensor.h for more detail (memcopy happens)
+   * @param[in] direction to transpose ex) 0:2:1
+   * @retval    LazyTensor *this
+   */
+  LazyTensor &transpose(std::string direction) ;
+
+  /**
+   * @brief     Wrapper method of sum. see tensor.h for more detail (memcopy happens)
+   * @param[in] direction to transpose ex) 0:2:1
```
Comment (dongju-chae) [inline nntrainer/include/lazy_tensor.h:107]: Typo?

File: test/unittest/unittest_nntrainer_lazy_tensor.cpp (hunk 1 above)
Comment (zhoonit) [reply test/unittest/unittest_nntrainer_lazy_tensor.cpp]: Fixed :) Thanks

File: nntrainer/include/lazy_tensor.h (hunk 2 above)
Comment (zhoonit) [reply nntrainer/include/lazy_tensor.h]: The line is deleted. Thank you!

**Comment by jijoongmoon** [comment]
There was CI error and it was fixed. please rebase.

**Comment by zhoonit** [comment]
> There was CI error and it was fixed. please rebase.

updated!

**Pull request merged**